javav2/example_code/ec2/src/main/java/com/example/ec2/Route53Scenario.java

```diff
@@ -0,0 +1,394 @@
+
+        } catch (Route53Exception e) {
+            System.err.println(e.getMessage());
+            System.exit(1);
+        }
+
+    }
+
+    public static void listDomainSuggestions(Route53DomainsClient route53DomainsClient, String domainSuggestion) {
+        try {
+            GetDomainSuggestionsRequest suggestionsRequest = GetDomainSuggestionsRequest.builder()
+                .domainName(domainSuggestion)
+                .suggestionCount(5)
+                .onlyAvailable(true)
+                .build();
+
+            GetDomainSuggestionsResponse response = route53DomainsClient.getDomainSuggestions(suggestionsRequest);
+            List<DomainSuggestion> suggestions = response.suggestionsList();
+            for (DomainSuggestion suggestion: suggestions) {
+                System.out.println("Suggestion Name: "+suggestion.domainName());
+                System.out.println("Availability: "+suggestion.availability());
+                System.out.println(" ");
+            }
+
+        } catch (Route53Exception e) {
+            System.err.println(e.getMessage());
+            System.exit(1);
+        }
+    }
+
+    public static void listPrices(Route53DomainsClient route53DomainsClient, String domainType) {
+        try {
+            ListPricesRequest pricesRequest = ListPricesRequest.builder()
+                .maxItems(10)
+                .tld(domainType)
+                .build();
+
+            ListPricesResponse response = route53DomainsClient.listPrices(pricesRequest);
+            List<DomainPrice> prices = response.prices();
+            for (DomainPrice pr: prices) {
+                System.out.println("Name: "+pr.name());
+                System.out.println("Registration: "+pr.registrationPrice().price() + " " +pr.registrationPrice().currency());
+                System.out.println("Renewal: "+pr.renewalPrice().price() + " " +pr.renewalPrice().currency());
+                System.out.println("Transfer: "+pr.transferPrice().price() + " " +pr.transferPrice().currency());
+                System.out.println("Transfer: "+pr.transferPrice().price() + " " +pr.transferPrice().currency());
+                System.out.println("Change Ownership: "+pr.changeOwnershipPrice().price() + " " +pr.changeOwnershipPrice().currency());
+                System.out.println("Restoration: "+pr.restorationPrice().price() + " " +pr.restorationPrice().currency());
+                System.out.println(" ");
+            }
+
+        } catch (Route53Exception e) {
+            System.err.println(e.getMessage());
+            System.exit(1);
+        }
+    }
+
+
+    public static void listBillingRecords(Route53DomainsClient route53DomainsClient) {
+        try {
+            Date currentDate = new Date();
+            LocalDateTime localDateTime = currentDate.toInstant().atZone(ZoneId.systemDefault()).toLocalDateTime();
+            ZoneOffset zoneOffset = ZoneOffset.of("+05:30");

```

Comment:
Is there a way to make this more generic, like to get the user's own offset instead of using a preset one? Or use the start of day?

Reply:
I will use US/Eastern